### PR TITLE
Update test commands

### PR DESCRIPTION
**Pull request by eoghanmurray**
May need some discussion

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -36,9 +36,12 @@
     "typescript": "^4.7.3"
   },
   "scripts": {
+    "build:all:lowmem": "NODE_OPTIONS='--max-old-space-size=4096' npm run build:all",
     "build:all": "yarn run concurrently --success=all -r -m=1 'yarn workspaces-to-typescript-project-references' 'yarn turbo run prepublish'",
     "test": "yarn run concurrently --success=all -r -m=1 'yarn workspaces-to-typescript-project-references --check' 'yarn turbo run test'",
     "test:watch": "yarn turbo run test:watch",
+    "test:update": "yarn turbo run test:update",
+    "reformat:all": "yarn turbo run reformat",
```
Comment (YunFeng0817) [inline package.json:44]:
```suggestion
    "format": "yarn prettier --write '**/*.{ts,md}'",
```
I think just one command like this is fine. It's not necessary to add one for all sub-packages.

File: package.json [hunk 2]
```diff
@@ -36,9 +36,12 @@
     "typescript": "^4.7.3"
   },
   "scripts": {
+    "build:all:lowmem": "NODE_OPTIONS='--max-old-space-size=4096' npm run build:all",
     "build:all": "yarn run concurrently --success=all -r -m=1 'yarn workspaces-to-typescript-project-references' 'yarn turbo run prepublish'",
```
Comment (YunFeng0817) [inline package.json:40]:
```suggestion
    "build:all": "NODE_OPTIONS='--max-old-space-size=4096' yarn run concurrently --success=all -r -m=1 'yarn workspaces-to-typescript-project-references' 'yarn turbo run prepublish'",
```

File: packages/rrweb/package.json
Comment (YunFeng0817) [inline packages/rrweb/package.json:1]: I feel like the old commands are enough for me.

File: packages/rrweb/package.json [hunk 3]
```diff
@@ -5,12 +5,17 @@
...
-    "test": "npm run bundle:browser && jest --testPathIgnorePatterns test/benchmark",
-    "test:headless": "PUPPETEER_HEADLESS=true npm run test",
-    "test:watch": "PUPPETEER_HEADLESS=true npm run test -- --watch",
+    "retest": "jest --testPathIgnorePatterns test/benchmark",
+    "build-and-test": "npm run bundle:browser && npm run retest",
+    "test": "npm run test:headless",
+    "test:headless": "PUPPETEER_HEADLESS=true npm run build-and-test",
+    "test:headful": "PUPPETEER_HEADLESS=false npm run build-and-test",
+    "test:watch": "npm run test:headless -- --watch",
+    "test:update": "npm run test:headless -- --updateSnapshot",
```
Comment (Juice10) [inline packages/rrweb/package.json:14]:
```suggestion
    "retest": "jest --testPathIgnorePatterns test/benchmark",
    "build-and-test": "yarn bundle:browser && yarn retest",
    "test:headless": "PUPPETEER_HEADLESS=true yarn build-and-test",
    "test:headful": "PUPPETEER_HEADLESS=false yarn build-and-test",
    "test": "yarn test:headless",
    "test:watch": "yarn test:headless -- --watch",
    "test:update": "yarn test:headless -- --updateSnapshot",
```

File: packages/rrweb/package.json [hunk 4]
```diff
@@ -5,12 +5,17 @@
...
     "repl": "npm run bundle:browser && node scripts/repl.js",
```
Comment (Juice10) [inline packages/rrweb/package.json:15]:
```suggestion
    "repl": "yarn bundle:browser && node scripts/repl.js",
```

File: packages/rrweb/package.json [hunk 5]
```diff
@@ -5,12 +5,17 @@
...
-    "live-stream": "yarn bundle:browser && node scripts/stream.js",
-    "dev": "yarn bundle:browser --watch",
+    "live-stream": "npm run bundle:browser && node scripts/stream.js",
+    "dev": "npm run bundle:browser --watch",
```
Comment (Juice10) [inline packages/rrweb/package.json:17]:
```suggestion
    "live-stream": "yarn bundle:browser && node scripts/stream.js",
    "dev": "yarn bundle:browser --watch",
```

File: turbo.json [hunk 6]
```diff
@@ -8,6 +8,8 @@
     },
     "test": {},
     "test:watch": {},
+    "test:update": {},
+    "reformat": {},
```
Comment (Juice10) [inline turbo.json:12]:
```suggestion
```

File: packages/types/package.json [hunk 7]
```diff
@@ -9,6 +9,7 @@
     "@rrweb/types"
   ],
   "scripts": {
+    "reformat": "yarn prettier --write '**/*.{ts,md}'",
```
Comment (Juice10) [inline packages/types/package.json:12]:
```suggestion
```

File: packages/rrweb/package.json [hunk 8]
```diff
@@ -5,12 +5,17 @@
...
+    "reformat": "yarn prettier --write '**/*.{ts,md}'",
```
Comment (Juice10) [inline packages/rrweb/package.json:18]:
```suggestion
```

File: packages/rrweb-snapshot/package.json [hunk 9]
```diff
@@ -7,9 +7,11 @@
     "prepack": "npm run bundle && npm run typings",
     "test": "jest",
     "test:watch": "jest --watch",
+    "test:update": "jest --updateSnapshot",
     "bundle": "rollup --config",
     "bundle:es-only": "cross-env ES_ONLY=true rollup --config",
-    "dev": "yarn bundle:es-only --watch",
+    "reformat": "yarn prettier --write '**/*.{ts,md}'",
```
Comment (Juice10) [inline packages/rrweb-snapshot/package.json:13]:
```suggestion
```

File: packages/rrweb-player/package.json [hunk 10]
```diff
@@ -28,11 +28,12 @@
   },
   "scripts": {
     "build": "rollup -c",
+    "reformat": "yarn prettier --write '**/*.{ts,md}'",
```
Comment (Juice10) [inline packages/rrweb-player/package.json:31]:
```suggestion
```

File: packages/rrvideo/package.json [hunk 11]
```diff
@@ -12,7 +12,7 @@
   "types": "build/index.d.ts",
   "scripts": {
     "build": "tsc",
-    "prepublish": "yarn build"
+    "prepublish": "npm run build"
```
Comment (Juice10) [inline packages/rrvideo/package.json:15]:
```suggestion
    "prepublish": "yarn build"
```

File: packages/rrdom/package.json [hunk 12]
```diff
@@ -19,6 +19,7 @@
     "url": "git+https://github.com/rrweb-io/rrweb.git"
   },
   "scripts": {
+    "reformat": "yarn prettier --write '**/*.{ts,md}'",
```
Comment (Juice10) [inline packages/rrdom/package.json:22]:
```suggestion
```

**Review by YunFeng0817 — APPROVED** [review]

**Pull request merged**